### PR TITLE
Change the APIs for eventing sources

### PR DESCRIPTION
**Pull request by houshengbo**
This PR changes the APIs for the the eventing sources, by adding the fields to enable/disable all the sources.

### CODE REVIEW

File: config/300-eventing.yaml [hunk 1]
```diff
@@ -99,6 +99,70 @@ spec:
...
+                    type: object
+                  ceph:
+                    description: Ceph settings
+                    properties:
+                      enabled:
+                        type: boolean
+                    type: object
+                  couchdb:
+                    description: Couchdb settings
+                    properties:
```
Comment (lionelvillard) [inline config/300-eventing.yaml:119]: `Couchdb` -> `Apache CouchDB`

File: config/300-eventing.yaml [hunk 2]
```diff
@@ -99,6 +99,70 @@ spec:
...
+                    type: object
+                  gitlab:
+                    description: Gitlab settings
+                    properties:
+                      enabled:
+                        type: boolean
+                    type: object
+                  kafka:
+                    description: Kafka settings
+                    properties:
```
Comment (lionelvillard) [inline config/300-eventing.yaml:137]: Kafka -> Apache Kafka

File: config/300-eventing.yaml [hunk 3]
```diff
@@ -99,6 +99,70 @@ spec:
...
+                        type: boolean
+                    type: object
+                  prometheus:
+                    description: Prometheus settings
+                    properties:
+                      enabled:
+                        type: boolean
+                    type: object
+                  rabbitmq:
+                    description: Rabbitmq settings
```
Comment (lionelvillard) [inline config/300-eventing.yaml:154]: Rabbitmq -> RabbitMQ

File: config/300-eventing.yaml [hunk 4]
```diff
@@ -99,6 +99,70 @@ spec:
                       description: The number of replicas that HA parts of the control plane will be scaled to
                       type: integer
                       minimum: 1
+              source:
+                description: The source configuration for Knative Eventing
+                properties:
+                  awssqs:
+                    description: Awssqs settings
```
Comment (lionelvillard) [inline config/300-eventing.yaml:106]: maybe Amazon SQS?

**Comment by matzew** [comment]
@lionelvillard @houshengbo do we have any plans for also shipping channels? (with their respective config)?

**Comment by matzew** [comment]
What's the reason to put this to the `KnativeEventing` CRD, rather than having some extra CRD? was there any feature track or so for this?

**Pull request merged**